Source/Fuse.Nodes/NodeGroup.uno

```diff
@@ -124,6 +137,7 @@ namespace Fuse
 		//used to prevent double-adding and double-removing if a derived class changes the UseContent
 		//state during rooting
 		bool _contentAdded;
+		MiniList<Node> _addedNodes;
```

Comment:
Did you consider simply using an array of nodes here instead? It seems trivial to compute the size known size here, and we can null it out when there's nothing added. Should be both small and simple...

```uno
var nodeCount = _nodes.Count + (UseTemplates ? _templates.Count : 0);
_addedNodes = new Node[nodeCount];
```

We can probably survive null-references in the list in the failure-case for non-node-templates.

Alternatively, we can just append, and at the end, if the actual size did not meet the computed size, we can shrink the list. That way we get ideal memory-allocation in the non-error-code, and don't have to deal with null-references in the rest of the code. But we pay a small price in the error-case, which I don't think is too bad.